### PR TITLE
Add transaction forwarder integration to ocr2vrf

### PR DESCRIPTION
**Pull request by vreff**
This PR integrates the transaction forwarder with OCR2VRF. 

- `NewOCRContractTransmitter` is not compatible with multiple sending keys, but it is used by potentially many ocr instances in-development. So, I have created `newContractTransmitterWithForwarder` and `NewTransmitterWithForwarder` that are compatible with the forwarder and don't interrupt current the current compatibility of `NewTransmitter`.

- Currently, the OCR job spec does only allows for the transmitterID to be a local sending key (see `core/services/job/orm.go`). I have added a check on the environment variable `EvmUseForwarders` to override this behavior.

- The implementation is automated, and adds all local sending keys to the job spec of a node. The node then picks up those sending keys, and uses them in a round-robin fashion to send transactions through the forwarder. This is somewhat brittle (you cannot add/remove the sending keys the job uses while the node is running), but seems to be the preferred approach.

### CODE REVIEW

File: core/services/job/orm.go [hunk 1]
```diff
@@ -230,9 +230,19 @@ func (o *orm) CreateJob(jb *Job, qopts ...pg.QOpt) error {
...
-					_, err := o.keyStore.Eth().Get(jb.OCR2OracleSpec.TransmitterID.String)
-					if err != nil {
-						return errors.Wrapf(ErrNoSuchTransmitterKey, "%v", jb.OCR2OracleSpec.TransmitterID)
+					chains := o.chainSet.Chains()
+					var useForwarders = true
+					for _, c := range chains {
+						if !c.Config().EvmUseForwarders() {
+							useForwarders = false
+						}
+					}
```
Comment (makramkd) [inline core/services/job/orm.go:239]: Interesting, so this doesn't switch on the `UseForwarder` field in the plugin config? This is a global config setting right?

File: core/services/job/orm.go [hunk 2]
```diff
@@ -230,9 +230,19 @@ func (o *orm) CreateJob(jb *Job, qopts ...pg.QOpt) error {
 			if jb.OCR2OracleSpec.TransmitterID.Valid {
 				switch jb.OCR2OracleSpec.Relay {
 				case relay.EVM:
-					_, err := o.keyStore.Eth().Get(jb.OCR2OracleSpec.TransmitterID.String)
-					if err != nil {
-						return errors.Wrapf(ErrNoSuchTransmitterKey, "%v", jb.OCR2OracleSpec.TransmitterID)
+					chains := o.chainSet.Chains()
+					var useForwarders = true
+					for _, c := range chains {
```
Comment (essamhassan) [inline core/services/job/orm.go:235]: Why do we have to check for all chains here? couldn't we get the chain from the spec and only check for that one chain?

**Review by essamhassan — DISMISSED** [review]
I would ask if we can have the logic for forwarder be blind integrated into the flow so we don't have to branch out a check whether it's forwarder flow or not like this:
1- On the ORM level, we have to check whether transmitterID OR []transmittersID is populated AND a valid address for the given chain

2- On the delegate level, we either accept transmitterID or []transmittersID
3- We end up always sending a list of addresses to the plugin and relayer runner
list.size >1 && effectiveTransmitter == list[0] should be a rejected path and always fail. We can't have more than 1 ocr address and no forwarder to send through


4- From here; transmitters will use your code of cycling through sending keys, this is always a list, in any case other than forwarder the list size has to be exactly 1.

File: core/services/job/orm.go (hunk 2 above)
Comment (vreff) [reply core/services/job/orm.go:235]: Changing this to use a `EVMChainForJob` function

File: core/services/ocr2/delegate.go [hunk 3]
```diff
@@ -125,6 +128,18 @@ func (d Delegate) ServicesForSpec(jobSpec job.Job) ([]job.ServiceCtx, error) {
...
+		ethSendingKeys, err2 := d.ethKs.GetAll()
+		if err2 != nil {
+			return nil, errors.Wrap(err2, "get eth sending keys")
+		}
+
+		// Automatically provide the node's local sending keys to the job spec.
+		for _, s := range ethSendingKeys {
+			sendingKeys = append(sendingKeys, s.Address.String())
+		}
+		spec.RelayConfig["sendingKeys"] = sendingKeys
```
Comment (makramkd) [inline core/services/ocr2/delegate.go:141]: Where do these updates to the relay config end up getting persisted? I see another update on line 154 but not sure where this ends up getting saved to the DB.

**Review by makramkd — COMMENTED** [review]
LGTM - are we using the forwarder in the ocr2vrf integration tests?

File: core/services/ocr2/delegate.go (hunk 3 above)
Comment (vreff) [reply core/services/ocr2/delegate.go:141]: I think it is used at runtime and not saved in the DB. @essamhassan is that correct?

**Comment by vreff** [comment]
> LGTM - are we using the forwarder in the ocr2vrf integration tests?

I'm gonna run stuff locally, then add any necessary changes. Once I know it's running locally I can add this flow to our integration test.

**Review by makramkd — APPROVED** [review]

**Review by makramkd — APPROVED** [review]

**Pull request merged**